dawsonc/recipes
# Adding & editing recipes

## Description

This PR adds the ability to add and edit recipes to the frontend. It also makes some minor changes to the backend.

## Related Issue

Closes:
- #5 
- #8

Partially resolves #7  but not really since authors are just string names in this PR (not linked to separate author objects).

## Changes Made

- TODO
- TODO

## How to Test

Backend changes are tested via new unit tests (`go test ./...`).

Frontend can be tested via:
1. Start the server
2. Load the app
3. Click on add a recipe
4. Enter the name of the recipe and your name
5. Click on "Add tag" and add a tag. Add another tag. Edit an existing tag.
6. Click on "add ingredient" and add an ingredient. Add another. Hit enter to add another. Edit an ingredient. Delete an ingredient.
7. Click on "add step" and add a step. Hit enter to add another step. Click the button to add another. Edit a step. Delete a step.
8. Click "create" to create or "cancel" to cancel. If you clicked create, there should be a new recipe available!

## Checklist

Please check the following items that apply to this pull request:

- [ ] The changes made are covered by unit tests and the CI tests are passing
- [ ] The code follows the project's coding style and conventions
- [ ] The changes made have been reviewed by another developer

## Additional Information

Please provide any additional information or context that might be useful to reviewers.
